subsys/zigbee/osif/zb_nrf_transceiver.c

```diff
@@ -257,332 +311,234 @@ zb_bool_t zb_trans_transmit(zb_uint8_t wait_type, zb_time_t tx_at,
 #endif
 	case ZB_MAC_TX_WAIT_NONE:
 		/* First transmit attempt without CCA. */
-		transmit_status = (zb_bool_t)nrf_802154_transmit_raw(
-			tx_buf, ZB_FALSE);
+		state_cache.radio_state = RADIO_802154_STATE_TRANSMIT;
+		err = radio_api->tx(radio_dev,
+				    IEEE802154_TX_MODE_DIRECT,
+				    NULL,
+				    &frag);
 		break;
 	default:
 		LOG_DBG("Illegal wait_type parameter: %d", wait_type);
 		ZB_ASSERT(0);
+		return ZB_FALSE;
+	}
+
+	state_cache.radio_state = RADIO_802154_STATE_RECEIVE;
```

Comment:
I'm not an expert here, but I feel that we have a bug here.
If we start the transmission directly from the SLEEP state, does the radio peripheral go to SLEEP or RECEIVE state after transmitting?
I really miss those state diagrams from radio driver API documentation.

Reply:
I'll try to get in debugging the actual state from radio driver. I also wrote to @hubertmis and asked for review of state caching.

Reply:
After TX radio driver automatically enters RX

Reply:
Does `radio_api->tx()` block until frame is transmitted?